src/BaGet.AWS/S3StorageService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.S3;
+using Amazon.S3.Model;
+using BaGet.AWS.Configuration;
+using BaGet.Core.Entities;
+using BaGet.Core.Services;
+using Microsoft.Extensions.Options;
+using NuGet.Versioning;
+
+namespace BaGet.AWS
+{
+    public class S3StorageService : IStorageService
+    {
+        private readonly string _bucket;
+        private readonly string _prefix;
+        private readonly AmazonS3Client _client;
+        private const string Seperator = "/";
+
+        public S3StorageService(IOptions<S3StorageOptions> options, AmazonS3Client client)
+        {
+            _bucket = options.Value.Bucket;
+            _prefix = options.Value.Prefix;
+            _client = client;
+
+            if (!string.IsNullOrEmpty(_prefix) && !_prefix.EndsWith(Seperator))
+                _prefix += Seperator;
+        }
+
+        private string PrepareKey(string path)
+        {
+            return _prefix + path.Replace("\\", Seperator);
+        }
+
+        public async Task<Stream> GetAsync(string path, CancellationToken cancellationToken = default)
+        {
+            MemoryStream stream = new MemoryStream();
+
+            try
+            {
+                GetObjectResponse res = await _client.GetObjectAsync(_bucket, PrepareKey(path), cancellationToken);
```

Comment:
This response should be disposed

Reply:
Consider referencing https://github.com/DotNetAnalyzers/IDisposableAnalyzers.